processing/src/main/java/org/apache/druid/query/timeseries/TimeseriesQueryQueryToolChest.java

```diff
@@ -138,7 +139,9 @@ public TimeseriesQueryQueryToolChest(TimeseriesQueryMetricsFactory queryMetricsF
 
       final Sequence<Result<TimeseriesResultValue>> finalSequence;
 
-      if (query.getGranularity().equals(Granularities.ALL) && !query.isSkipEmptyBuckets()) {
+      if (query.getGranularity().equals(Granularities.ALL) &&
```

Comment:
To be more sql-compliant, the timeseries query returns something if there is no grouping key even when there is nothing to return. However, this behavior is not preferred I think when `bySegment` is set because, `bySegment` is used for caching in historicals or debugging where exact results are supposed to be returned. When caching is enabled, it will populate empty result in historicals after this change, but the broker will return the correct result since `bySegment` won't be set for it. I added some comment for it.